src/HTTPClient.ts

```diff
@@ -27,10 +27,41 @@ export class HTTPClient {
 	 * @public
 	 * @constructor
 	 * Creates an instance of a HTTPClient
-	 * @param {Middleware} middleware - The first middleware of the middleware chain
+	 * @param {Middleware| Middleware[]} middleware - The first middleware of the middleware chain or the array of middleware handlers
 	 */
-	public constructor(middleware: Middleware) {
-		this.middleware = middleware;
+	public constructor(middleware: Middleware | Middleware[]) {
+		this.setMiddleware(middleware);
+	}
+
+	/**
+	 * @private
+	 * Verifies if the middleware passed is an array or not before setting this.middleware property
+	 * @param {any} middleware - The middleware passed
+	 * @returns Nothing
+	 */
+
+	private setMiddleware(middleware: any): void {
```

Comment:
```suggestion
	private setMiddleware(...middleware: Middleware[]): void {
```

Reply:
This change could effect public interfaces. Will need to look into this separately to apply this with minimal changes to existing code.

Reply:
as reviewed on the call, not necessarily, it could but if we want to avoid any public API change we could move up the type test.

Reply:
The type test is in private setMiddleware function so that the signature of public constructor of Client.ts is not changed.